src/components/legend.ts

```diff
@@ -280,6 +282,7 @@ export module Components {
       entries.select("path").attr("d", (d: any, i: number, j: number) => this.symbol()(d, j)(layout.textHeight * 0.6))
                             .attr("transform", "translate(" + (layout.textHeight / 2) + "," + layout.textHeight / 2 + ")")
                             .attr("fill", (value: string) => this._colorScale.scale(value))
+                            .attr("opacity", (d: any, i: number) => this.symbolOpacity()(d, i))
```

Comment:
If you inspect the elements in the third legend in the plunker I created, an opacity based on the index works:

```
legend.symbolOpacity(function(d,i) {return 1 - i/5;})
```

I can codify this up into a unit test (the current unit test has a function based on datum).